src/main/java/org/spongepowered/common/mixin/realtime/mixin/MixinMinecraftServer.java

```diff
@@ -34,17 +34,17 @@
 @Mixin(MinecraftServer.class)
 public abstract class MixinMinecraftServer implements IMixinMinecraftServer {
 
-    private static long lastTickMillis = System.currentTimeMillis();
+    private static long lastTickNanos = System.nanoTime();
     private static long realTimeTicks = 1;
 
     @Inject(method = "tick", at = @At("HEAD"))
     public void tick(CallbackInfo ci) {
-        long currentMillis = System.currentTimeMillis();
-        realTimeTicks = (currentMillis - lastTickMillis) / 50;
+        long currentNanos = System.nanoTime();
+        realTimeTicks = (currentNanos - lastTickNanos) / 50000000;
```

Comment:
Perhaps a constant initialized as `TimeUnit.SECONDS.toNanos(1) / 20` would be less magic?